include/hybridmap/HybridNeutralAtomMapper.hpp

```diff
@@ -0,0 +1,534 @@
+   * OpenQASM
+   * @return The mapped quantum circuit with native AOD operations
+   */
+  [[maybe_unused]] std::string getMappedQcAOD() {
+    std::stringstream ss;
+    this->mappedQcAOD.dumpOpenQASM(ss, false);
+    return ss.str();
+  }
+
+  /**
+   * @brief Saves the mapped quantum circuit with AOD operations to a file.
+   * @param filename The name of the file to save the mapped quantum circuit
+   * with AOD operations to
+   */
+  [[maybe_unused]] void saveMappedQcAOD(const std::string& filename) {
+    std::ofstream ofs(filename);
+    this->mappedQcAOD.dumpOpenQASM(ofs, false);
+  }
+
+  /**
+   * @brief Schedules the mapped quantum circuit on the neutral atom
+   * architecture.
+   * @details For each gate/operation in the input circuit, the scheduler checks
+   * the earliest possible time slot for execution. If the gate is a multi qubit
+   * gate, also the blocking of other qubits is taken into consideration. The
+   * execution times are read from the neutral atom architecture.
+   * @param verboseArg If true, prints additional information
+   * @param createAnimationCsv If true, creates a csv file for the animation
+   * @param shuttlingSpeedFactor The factor to speed up the shuttling time
+   * @return The results of the scheduler
+   */
+  [[maybe_unused]] SchedulerResults schedule(bool verboseArg           = false,
+                                             bool createAnimationCsv   = false,
+                                             fp   shuttlingSpeedFactor = 1.0) {
+    scheduler = NeutralAtomScheduler(arch);
+    return scheduler.schedule(mappedQcAOD, hardwareQubits.getInitialHwPos(),
+                              verboseArg, createAnimationCsv,
+                              shuttlingSpeedFactor);
+  }
+
+  /**
+   * @brief Saves the animation csv file of the scheduled quantum circuit.
+   * @return The animation csv string
+   */
+  [[maybe_unused]] std::string getAnimationCsv() {
+    return scheduler.getAnimationCsv();
+  }
+
+  /**
+   * @brief Saves the animation csv file of the scheduled quantum circuit.
+   * @param filename The name of the file to save the animation csv file to
+   */
+  [[maybe_unused]] void saveAnimationCsv(const std::string& filename) {
+    scheduler.saveAnimationCsv(filename);
+  }
+
+  /**
+   * @brief Converts a mapped circuit down to the AOD level and CZ level.
+   * @details SWAP gates are decomposed into CX gates. Then CnX gates are
+   * decomposed into CnZ gates. Move operations are combined if possible and
+   * then converted into native AOD operations.
+   * @param qc The already mapped quantum circuit with abstract SWAP gates and
+   * MOVE operations
+   * @return The mapped quantum circuit with native AOD operations
+   */
+  QuantumComputation convertToAod(qc::QuantumComputation& qc);
+
+  [[maybe_unused]] [[nodiscard]] std::map<HwQubit, HwQubit>
+  getInitHwPos() const {
+    return hardwareQubits.getInitialHwPos();
+  }
```

Comment:
if unused, this can also be removed IMHO

Reply:
I use these functions for the python bindings, so Clion does not detect the use and therefore I added the [[maybe_unused]] part.

Reply:
I see. That kind of makes sense. Then only one small comment here: Consider using the same name for the function as in the other class. Consistency is key. Right now, this is using `getInitHwPos` and `getInitialHwPos`.

Reply:
You are right, didn't notice that.